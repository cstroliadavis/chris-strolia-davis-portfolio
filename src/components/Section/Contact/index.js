import './Contact.css';
import { useState } from "react";
import { validateEmail } from '../../../utils/string-helpers';
import { contactUs } from "../../../services/contact";

const pause = (millis) => new Promise((resolve) => setTimeout(resolve, millis));

const defaultFormFields = { name: '', email: '', message: '' };
const defaultErr = {
  showNameRequired: '',
  showEmailRequired: '',
  showEmailMustBeValid: '',
  showMessageRequired: '',
};

export default function Contact() {
  const [ formState, setFormState ] = useState(defaultFormFields)
  const [ messageSent, setMessageSent ] = useState('');
  const [ err, setErr ] = useState(defaultErr);
  const [ submitDisabled, setSubmitDisabled ] = useState(true);
  const { name, email, message } = formState;

  function formNotReady({ name, email, message }) {
    return !(name && email && validateEmail(email) && message);
  }

  function cleanErrors({ name, email, message }) {
    const cleared = {}
    if (name) cleared.showNameRequired = '';
    if (email) cleared.showEmailRequired = '';
    if (validateEmail(email)) cleared.showEmailMustBeValid = '';
    if (message) cleared.showMessageRequired = '';

    setErr({ ...err, ...cleared });
  }

  const validations = {
    email(value) {
      setErr({
        ...err,
        showEmailRequired: value ? '':'show',
        showEmailMustBeValid: value && !validateEmail(value) ? 'show':'',
      });
    },
    name(value) {
      setErr({ ...err, showNameRequired: value ? '':'show' });
    },
    message(value) {
      setErr({ ...err, showMessageRequired: value ? '':'show' });
    },
  };

  function modifyInput(e) {
    const { name, value } = e.target;
    const newVals = { ...formState, [name]: value };
    cleanErrors(newVals);
    setFormState(newVals);
    setSubmitDisabled(formNotReady(newVals));
  }

  const inputEvents = {
    onBlur(e) {
      const { name: target, value } = e.target;

      const action = validations[target];
      action(value);

      setSubmitDisabled(formNotReady({ name, email, message }))
    },
    onChange: modifyInput,
    onInput: modifyInput,
  }

  function handleSubmit(e) {
    e.preventDefault();
    contactUs(formState)
      .then(() => setMessageSent('show'))
      .then(() => pause(5000))
      .then(() => setMessageSent(''));
  }

  return (
    <div className="container">
      <h1>Contact me</h1>
      <form id="contact-form" onSubmit={ handleSubmit }>
        <div className="name">
          <input type="text" name="name" placeholder="Name" defaultValue={ name } { ...inputEvents }/>
          <div className={ `error ${ err.showNameRequired }` }>Please enter a name.</div>
        </div>
        <div className="email">
          <input type="email" name="email" placeholder="E-mail" defaultValue={ email } { ...inputEvents }/>
          <div className={ `error ${ err.showEmailRequired }` }>Please enter an email.</div>
          <div className={ `error ${ err.showEmailMustBeValid }` }>Please enter a valid email address.</div>
        </div>
        <div className="message">
          <textarea name="message" placeholder="Message" rows="5" defaultValue={ message } { ...inputEvents }/>
          <div className={ `error ${ err.showMessageRequired }` }>Please include a message.</div>
        </div>
        <div className={ `message-sent ${ messageSent }` }>Message was sent successfully.</div>
        <button disabled={ submitDisabled } type="submit">Submit</button>
      </form>
    </div>
  )
}
