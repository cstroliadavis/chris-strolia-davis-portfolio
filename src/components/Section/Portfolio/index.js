import loading from '../../../assets/img/loading-gif.gif';
import './Portfolio.css';
import { useEffect, useState } from "react";
import AppCard from '../../AppCard';
import { getAllApplications } from '../../../services/applications.repository';

export default function Portfolio() {
  const [ isLoading, setIsLoading ] = useState(false);
  const [ applications, setApplications ] = useState([]);
  const [ errors, setErrors ] = useState(null);

  useEffect(() => {
    setIsLoading(true);

    getAllApplications()
      .then((appData) => setApplications(appData))
      .catch(setErrors)
      .finally(() => setIsLoading(false))

    return () => {
    };
  }, []);

  return (
    <div className="container">
      {
        isLoading
          ? <img alt="Loading" src={ loading }/>
          :errors
            ? <div className="error">There was an issue loading data. Please try again</div>
            :<div className="app-list">
              { applications.map(app => (<AppCard { ...app } key={ app.id }/>)) }
            </div>
      }
    </div>
  );
}
