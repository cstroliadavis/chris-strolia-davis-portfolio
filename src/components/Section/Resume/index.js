import './Resume.css';

export default function Resume() {
  return (
    <div className="container">
      <h1>Résumé</h1>
      <a href="./Christopher-Strolia-Davis-Resume-online.pdf" className="download">
        <img src="./9-2-resume-picture.png" alt="Chris Strolia-Davis Résumé"/>
      </a>
      <h2>Proficiencies</h2>
      <div className="proficiencies">
        <ul>
          <li><h3>Recent</h3>
            <ul>
              <li>Node JS, express, knex, Deno, Typescript, Postgres, redis, elasticsearch, Docker, nginx,
                Lerna, Swagger, yarn, make, mocha, jest, sinon, chai, eslint, babel, git, gitlab (including gitlab ci),
                github, bitbucket, jira, confluence, Datadog, AWS (S3, EC2, OpenSearch, Cognito, IAM), React, Vue (+
                vuex), Handlebars, HTML, CSS, JavaScript, SQL
              </li>
            </ul>
          </li>
          <li><h3>Additional</h3>
            <ul>
              <li>C#, VB.Net, ASP.Net, Linq, Entity Framework, jQuery, XML, XSLT, XPath, XSD, MS Access,
                SQL Server, Oracle, IBM DB2, Teradata, MySQL, Mongo, Raven, MS Excel, Crystal Reports (Business
                Objects), SQL Server Reporting Services (SSRS), SQL Server Integration Services (SSIS), Data
                Transformation Services (DTS), Team Foundation Services (TFS), Subversion, CVS, MS Build, Exchange Web
                Services (EWS), Java, Spring Framework, knockout JS, grunt, gulp, webpack, rollup, browserify, Angular
                JS, ... and more I’ve forgotten about.
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  );
}
