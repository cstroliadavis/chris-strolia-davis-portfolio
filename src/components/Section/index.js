import About from "./About";

const sections = {
  about: About
}

export default function Section({ currentSection }) {
  function Default() {
    return (<>{ currentSection.title }</>);
  }

  const View = sections[currentSection.name] ?? Default;

  return (
    <main className={`section ${currentSection.name}`}>
      <View />
    </main>
  );
}
