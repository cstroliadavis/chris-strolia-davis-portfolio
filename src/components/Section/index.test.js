import { render, cleanup } from '@testing-library/react';
import Section from './index'

afterEach(cleanup);

describe('Section', () => {
  it('renders', () => {
    render(<Section currentSection={{ name: 'about' }} />)
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Section currentSection={{ name: 'about' }}/>);

    expect(asFragment()).toMatchSnapshot();
  });

  it('handles when section does not exist', () => {
    const { getByText } = render(<Section currentSection={{ name: 'stuff', title: 'Stuff n Things' }}/>);

    expect(getByText('Stuff n Things')).toBeTruthy();
  });
});
