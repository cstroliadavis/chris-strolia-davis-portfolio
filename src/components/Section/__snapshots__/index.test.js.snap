// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="section about"
  >
    <img
      alt="Portrait of Chris Strolia-Davis"
      class="owner-portrait"
      src="CSD-Profile.png"
    />
    <p
      class="owner-bio"
    >
      Chris has been developing high class, enterprise level, secure, efficient, scalable software for over two decades. In that time, he has had the opportunity to work with numerous different technologies in various large organizations. This has brought him a plethora of experience, not only with these tools and technologies, but also with various different types of people, environments and challenges associated with high quality, enterprise level development. As a result, he has learned to adapt quickly to changing environments, and more importantly, to work on elevating teamwork and team building within those areas as this has always had the most dramatic and long lasting positive effects in all circumstances. Chris regularly mentors other developers, helping them to more easily understand complex systems and processes, but also to understand why these systems and processes are in place and how they benefit the team and the organization. Chris is focused on continuous improvement, not only for himself, but for the teams and organizations in which he works. While Chris has learned that technologies come and go, and he can quickly adapt to them as they do, he has maintained a focus on Javascript and SQL based technologies for the bulk of his career and has a keen understanding of these technologies above all others.
    </p>
  </main>
</DocumentFragment>
`;
